graphite_raintank.py

```diff
@@ -180,66 +180,17 @@ def search_series(self, query):
         return dict(leafs=leafs, branches=branches)
 
     def fetch_multi(self, nodes, start_time, end_time):
-        step = None
-        node_ids = {}
-        for node in nodes:
-            for metric in node.reader.metrics:
-                if step is None or metric.interval < step:
-                    step = metric.interval
-
         with statsd.timer("graphite-api.fetch.raintank_query.query_duration"):
             data = self.fetch_from_tank(nodes, start_time, end_time)
         series = {}
-        delta = None
+        step = None
         with statsd.timer("graphite-api.fetch.unmarshal_raintank_resp.duration"):
+            for path, arr in data.iteritems():
+                series[path] = [p[0] for p in arr[1]]
+                if step is None or step < arr[0]:
+                    step = arr[0]
 
-            for path, points in data.iteritems():
-                datapoints = []
-                next_time = start_time;
-                
-                max_pos = len(points)
-
-                if max_pos == 0:
-                    for i in range(int((end_time - start_time) / step)):
-                        datapoints.append(None)
-                    series[path] = datapoints
-                    continue
-
-                pos = 0
-
-                if delta is None:
-                    delta = (points[0][1] % start_time) % step
-                    # ts[0] is always greater then start_time.
-                    if delta == 0:
-                        delta = step
-
-                while next_time <= end_time:
-                    # check if there are missing values from the end of the time window
-                    if pos >= max_pos:
-                        datapoints.append(None)
-                        next_time += step
-                        continue
-
-                    ts = points[pos][1]
-                    # read in the metric value.
-                    v = points[pos][0]
-
-                    # pad missing points with null.
-                    while ts > (next_time + step):
-                        datapoints.append(None)
-                        next_time += step
-
-                    datapoints.append(v)
-                    next_time += step
-                    pos += 1
-                    if (ts + step) > end_time:
-                        break
-
-                series[path] = datapoints
-
-        if delta is None:
-            delta = 1
-        time_info = (start_time + delta, end_time, step)
+        time_info = (start_time + 1, end_time, step)
```

Comment:
The data returned to graphite does not include any timestamps, it is just an array of points.  The start_time value in the time_info tuple that is returned tells graphite what the TS of the first point is.  So setting this to just `start_time+1` is not correct. The TS would be somewhere in the range `start_time < TS <= (start_time + step)`

However, if there are metrics from many series, that are collected on different time offsets, then this doesnt really matter.   So maybe we just set it to `(start_time +1) + ((start_time + 1) % step)`  ie, on the step boundary

eg. if the start_time is 100, and the step is 10, then the first point gets set to 110.  In reality, the first point would be 100 < point <= 110

This reduces the accuracy of the data being graphed, which is more significant when the interval is higher. Ie data collected every 1 minute would always be displayed on the 1minute time boundary, regardless of when during the minute it was collected.